includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,512 @@
+		 *         @type string|boolean $requires_php   Minimum version of PHP required by the theme. False if all versions are supported.
+		 *         @type string         $download_link  A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = array_filter( (array) apply_filters( 'amp_reader_themes', $themes ) );
+
+		foreach ( $themes as &$theme ) {
+			$theme['availability'] = $this->get_theme_availability( $theme );
+		}
+
+		$this->themes = $themes;
+
+		return $this->themes;
+	}
+
+	/**
+	 * Gets a reader theme by slug.
+	 *
+	 * @param string $slug Theme slug.
+	 * @return array Theme data.
+	 */
+	public function get_reader_theme( $slug ) {
+		return current(
+			array_filter(
+				$this->get_themes(),
+				static function ( $theme ) use ( $slug ) {
+					return $theme['slug'] === $slug;
+				}
+			)
+		);
+	}
+
+	/**
+	 * Retrieves theme data.
+	 *
+	 * @param boolean $from_api Whether to return theme data from the wordpress.org API. Default false.
+	 * @return array Theme data from the wordpress.org API.
+	 */
+	public function get_default_supported_reader_themes( $from_api = false ) {
+		if ( ! $from_api ) {
+			$themes = self::DEFAULT_READER_THEMES;
+
+			return $themes;
+		}
+
+		// Note: This can be used to refresh the hardcoded raw theme data.
+		if ( ! function_exists( 'themes_api' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/theme.php';
+		}
+
+		$response = themes_api(
+			'query_themes',
+			[
+				'author'   => 'wordpressdotorg',
+				'per_page' => 24, // There are only 12 as of 05/2020.
+			]
+		);
+
+		if ( ! $response || is_wp_error( $response ) ) {
+			return [];
+		}
+
+		if ( is_array( $response ) ) {
+			$response = (object) $response;
+		}
+
+		$supported_themes = array_diff(
+			AMP_Core_Theme_Sanitizer::get_supported_themes(),
+			[ 'twentyten' ] // Excluded because not responsive.
+		);
+
+		$supported_themes_from_response = array_filter(
+			$response->themes,
+			static function ( $theme ) use ( $supported_themes ) {
+				return in_array( $theme->slug, $supported_themes, true );
+			}
+		);
+
+		return $supported_themes_from_response;
+	}
+
+	/**
+	 * Prepares a single theme.
+	 *
+	 * @param array|object $theme Theme data from the wordpress.org themes API.
+	 * @return array|object Prepared theme array.
+	 */
+	public function prepare_theme( $theme ) {
+		$theme_array = (array) $theme;
+
+		$keys = [
+			'name',
+			'slug',
+			'preview_url',
+			'screenshot_url',
+			'homepage',
+			'description',
+			'requires',
+			'requires_php',
+			'download_link',
+		];
+
+		$prepared_theme = array_filter(
+			$theme_array,
+			static function ( $key ) use ( $keys ) {
+				return in_array( $key, $keys, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		foreach ( $keys as $key ) {
+			if ( ! array_key_exists( $key, $prepared_theme ) ) {
+				$prepared_theme[ $key ] = '';
+			}
+		}
```

Comment:
```suggestion
		$prepared_theme = array_merge(
			array_fill_keys( $keys, '' ),
			wp_array_slice_assoc( $theme_array, $keys )
		);
```

Reply:
Nice. I thought there might be a shorter way to do this than the `foreach` loop, but this is another level 😎